### PR TITLE
Efrs 761 update organization api

### PR DESCRIPTION
**Pull request by omorgunova-pl**

### CODE REVIEW

**Review by isobolev — COMMENTED** [review]
please merge the latest develop into this branch, there are some conflicts. Add reviewer for Java part

File: ui/src/app/features/app-user-list/application-user-list-facade.ts [hunk 1]
```diff
@@ -45,45 +40,38 @@ export class ApplicationUserListFacade implements IFacade {
...
   private sub: Subscription;
+  userGlobalRole$: Observable<Role>;
 
   constructor(private store: Store<AppState>, private userService: AppUserService) {
     this.appUsers$ = store.select(selectAppUsers);
-    this.availableEmails$ = combineLatest(
-      [store.select(selectUsers),
-      this.appUsers$]
-    ).pipe(
+    this.availableEmails$ = combineLatest([store.select(selectUsers), this.appUsers$]).pipe(
```
Comment (isobolev) [inline ui/src/app/features/app-user-list/application-user-list-facade.ts:52]: this.store

File: ui/src/app/features/app-user-list/application-user-list-facade.ts [hunk 2]
```diff
@@ -45,45 +40,38 @@ export class ApplicationUserListFacade implements IFacade {
...
     );
-    this.organizationRole$ = this.store.select(selectUserRollForSelectedOrganization);
-    this.userRole$ = combineLatest(
-      [this.store.select(selectUserRollForSelectedApp),
-        this.organizationRole$]
-    ).pipe(
-      map(([applicationRole, organizationRole]) => {
-        // the organization role (if OWNER or ADMINISTRATOR) should prevail on the application role
-        return organizationRole !== Role.USER ? organizationRole : applicationRole;
+    this.userGlobalRole$ = store.select(selectCurrentUserRole);
```
Comment (isobolev) [inline ui/src/app/features/app-user-list/application-user-list-facade.ts:61]: this.store

File: ui/src/app/features/app-user-list/application-user-list-facade.ts [hunk 3]
```diff
@@ -95,67 +83,61 @@ export class ApplicationUserListFacade implements IFacade {
...
+    this.store.dispatch(
+      deleteUserFromApplication({
+        applicationId: this.selectedApplicationId,
+        userId,
+      })
+    );
   }
 
-  unsubscribe(): void {
-    this.sub.unsubscribe();
```
Comment (isobolev) [inline ui/src/app/features/app-user-list/application-user-list-facade.ts:159]: memory leak. please put back this.sub.unsubscribe();

File: ui/src/app/features/application-list/application-list-facade.ts [hunk 4]
```diff
@@ -14,62 +14,37 @@
...
+import { AppState } from 'src/app/store';
+import { createApplication, loadApplications } from 'src/app/store/application/action';
+import { selectApplications, selectIsPendingApplicationList } from 'src/app/store/application/selectors';
 
 @Injectable()
 export class ApplicationListFacade implements IFacade {
   applications$: Observable<Application[]>;
-  selectedOrganizationId$: Observable<string>;
   isLoading$: Observable<boolean>;
   userRole$: Observable<string>;
```
Comment (isobolev) [inline ui/src/app/features/application-list/application-list-facade.ts:30]: userRole is not used

File: ui/src/app/features/application-list/application-list-facade.ts [hunk 5]
```diff
@@ -14,62 +14,37 @@
...
-  unsubscribe(): void {
-    this.selectedOrganizationIdSubscription.unsubscribe();
-  }
-
-  getOrgId(): string {
-    return this.selectedOrgId;
+    this.store.dispatch(createApplication({ name }));
   }
 
+  unsubscribe(): void {}
```
Comment (isobolev) [inline ui/src/app/features/application-list/application-list-facade.ts:49]: a redundant method, there are no subscriptions to unsubscribe from

File: ui/src/app/features/model-list/model-list-facade.ts [hunk 6]
```diff
@@ -23,85 +23,77 @@ import { AppState } from 'src/app/store';
...
 
   private currentArgsAndApplicationSubscription: Subscription;
-  selectedOrganizationId: string;
   selectedApplicationId: string;
 
   constructor(private store: Store<AppState>) {
     this.models$ = store.select(selectModels);
     this.isLoading$ = store.select(selectPendingModel);
-    this.selectedOrganization$ = store.select(selectCurrentOrganizationId);
+    this.userRole$ = store.select(selectCurrentUserRole);
```
Comment (isobolev) [inline ui/src/app/features/model-list/model-list-facade.ts:43]: this.userRole$ is initialized twice. this one will be overwritten on line 45

File: ui/src/app/features/model-list/model-list-facade.ts [hunk 7]
```diff
@@ -23,85 +23,77 @@ import { AppState } from 'src/app/store';
...
+    this.store.dispatch(
+      deleteModel({
+        applicationId: this.selectedApplicationId,
+        modelId,
+      })
+    );
   }
 
-  unsubscribe(): void {
-    this.currentArgsAndApplicationSubscription.unsubscribe();
```
Comment (isobolev) [inline ui/src/app/features/model-list/model-list-facade.ts:105]: memory leak, please put back

File: ui/src/app/features/user-list/user-list-facade.ts [hunk 8]
```diff
@@ -22,105 +22,78 @@ import { AppUser } from 'src/app/data/interfaces/app-user';
...
   availableRoles$: Observable<string[]>;
   isLoading$: Observable<boolean>;
   currentUserId$: Observable<string>;
   currentUserEmail$: Observable<string>;
-  userRole$: Observable<string>;
+  userRole$: Observable<any>;
 
-  private selectedOrganization: string;
+  private sub: Subscription;
+  userspage$: Observable<any>;
```
Comment (isobolev) [inline ui/src/app/features/user-list/user-list-facade.ts:42]: unused declaration

File: ui/src/app/features/user-list/user-list-facade.ts [hunk 9]
```diff
@@ -22,105 +22,78 @@ import { AppUser } from 'src/app/data/interfaces/app-user';
...
   users$: Observable<AppUser[]>;
   availableRoles$: Observable<string[]>;
   isLoading$: Observable<boolean>;
   currentUserId$: Observable<string>;
   currentUserEmail$: Observable<string>;
-  userRole$: Observable<string>;
+  userRole$: Observable<any>;
 
-  private selectedOrganization: string;
+  private sub: Subscription;
```
Comment (isobolev) [inline ui/src/app/features/user-list/user-list-facade.ts:41]: unused declaration

File: ui/src/app/features/user-list/user-list-facade.ts [hunk 10]
```diff
@@ -22,105 +22,78 @@ import { AppUser } from 'src/app/data/interfaces/app-user';
...
 export class UserListFacade implements IFacade {
-  selectedOrganization$: Observable<string>;
-  selectedOrganizationName$: Observable<string>;
   users$: Observable<AppUser[]>;
   availableRoles$: Observable<string[]>;
   isLoading$: Observable<boolean>;
   currentUserId$: Observable<string>;
   currentUserEmail$: Observable<string>;
-  userRole$: Observable<string>;
+  userRole$: Observable<any>;
```
Comment (isobolev) [inline ui/src/app/features/user-list/user-list-facade.ts:39]: `Observable<string>;`

File: ui/src/app/features/user-list/user-list-facade.ts [hunk 11]
```diff
@@ -22,105 +22,78 @@ import { AppUser } from 'src/app/data/interfaces/app-user';
...
-  private selectedOrganizationSubscription: Subscription;
-
-  constructor(
-    private store: Store<AppState>,
-    private userService: UserService,
-  ) {
-    this.selectedOrganization$ = store.select(selectCurrentOrganizationId);
-    this.selectedOrganizationName$ = store.select(selectSelectedOrganization).pipe(map(org => org.name));
+  constructor(private store: Store<AppState>, private userService: UserService) {
     this.users$ = store.select(selectUsersWithOwnerApp);
```
Comment (isobolev) [inline ui/src/app/features/user-list/user-list-facade.ts:45]: this.store

File: ui/src/app/features/user-list/user-list-facade.ts [hunk 12]
```diff
@@ -22,105 +22,78 @@ import { AppUser } from 'src/app/data/interfaces/app-user';
...
 
-  private selectedOrganizationSubscription: Subscription;
-
-  constructor(
-    private store: Store<AppState>,
-    private userService: UserService,
-  ) {
-    this.selectedOrganization$ = store.select(selectCurrentOrganizationId);
-    this.selectedOrganizationName$ = store.select(selectSelectedOrganization).pipe(map(org => org.name));
+  constructor(private store: Store<AppState>, private userService: UserService) {
```
Comment (isobolev) [inline ui/src/app/features/user-list/user-list-facade.ts:44]: userService is not used?

File: ui/src/app/features/user-list/user-list-facade.ts [hunk 13]
```diff
@@ -22,105 +22,78 @@ import { AppUser } from 'src/app/data/interfaces/app-user';
...
   }
 
   loadAvailableRoles(): void {
     this.store.dispatch(loadRolesEntityAction());
   }
 
-  unsubscribe(): void {
-    this.selectedOrganizationSubscription.unsubscribe();
-  }
+  unsubscribe(): void {}
```
Comment (isobolev) [inline ui/src/app/features/user-list/user-list-facade.ts:98]: redundant method

File: ui/src/app/pages/organization/organization.component.ts [hunk 14]
```diff
@@ -21,11 +21,10 @@ import { OrganizationService } from './organization.service';
   selector: 'app-organization',
```
Comment (isobolev) [inline ui/src/app/pages/organization/organization.component.ts:21]: looks like this component can be removed

File: ui/src/app/pages/organization/organization.service.ts [hunk 15]
```diff
@@ -32,23 +32,22 @@ export class OrganizationService {
```
Comment (isobolev) [inline ui/src/app/pages/organization/organization.service.ts:32]: is this service is used anywhere? should be removed

File: ui/src/app/store/user/reducers.ts [hunk 16]
```diff
@@ -14,45 +14,51 @@
...
+  updateUserRoleFailAction,
+  updateUserRoleSuccessAction,
 } from './action';
 
-import {EntityState, createEntityAdapter, EntityAdapter} from '@ngrx/entity';
+import { EntityState, createEntityAdapter, EntityAdapter } from '@ngrx/entity';
 
 export interface AppUserEntityState extends EntityState<AppUser> {
   isPending: boolean;
+  currentId: string | null;
```
Comment (isobolev) [inline ui/src/app/store/user/reducers.ts:32]: this property is never set

File: ui/src/app/pages/organization/organization.component.ts (hunk 14 above)
Comment (omorgunova-pl) [reply ui/src/app/pages/organization/organization.component.ts:21]: there will be a separate task for organization module deleting

File: ui/src/app/pages/organization/organization.service.ts (hunk 15 above)
Comment (omorgunova-pl) [reply ui/src/app/pages/organization/organization.service.ts:32]: there will be a separate task for organization module deleting

File: ui/src/app/features/user-list/user-list-facade.ts (hunk 11 above)
Comment (isobolev) [reply ui/src/app/features/user-list/user-list-facade.ts:41]: not resolved everywhere

**Review by isobolev — APPROVED** [review]

**Pull request merged**